src/snapshot/metadata.rs

```diff
@@ -144,6 +146,15 @@ impl From<ScriptContextType> for InstanceContext {
     }
 }
 
+impl From<Option<ScriptContextType>> for InstanceContext {
+    fn from(script_type: Option<ScriptContextType>) -> Self {
+        Self {
+            script_type: script_type.unwrap_or_default(),
+            ..Self::new()
+        }
+    }
+}
+
```

Comment:
I wasn't sure if I should keep this in for the tests. Let me know your thoughts.